README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Get the author for this document
+res.headers.get('earthstar-author') === '@bbbb.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+
+// Get the content of the document back out.
+// If it's JSON, you should use `.json()`
+const content = await res.text()
+```
+
+GET an identity 'name' - Generate an identity
+
+```JavaScript
+// Specify `@XXXX` where `X` is a lowercase ascii character
+// This will generate and store a keypair for this identity
+const res = fetch('earthstar://@xmpl')
+
+// Get the generated address for your identity
+const {address} = await res.json()
+
+address === '@xmpl.xxxxxxx'
+```
+
+POST some data, sign with identity
+
+```JavaScript
+// It's good practice to use a file extension so that the proper mime type gets set
+const res = await fetch('earthstar://+gardening.xxxxx/example.json', {
+  method: 'post',
+  body: JSON.stringify({
+    hello: 'world'
+  }),
+  headers: {
+    // Use the identity address you got before to sign your update
+    "earthstar-author": "@xmpl.xxxxxx"
```

Comment:
Aha, so here we're asking the browser to remember the secret it generated before to complete the keypair?

Hmm, we'll need a way for users to add their existing keypairs into the system too, as well as generating new ones.

Reply:
Yeah, maybe we could also have an `"earthstar-secret": "HEX STRING HERE"` header for manually supplying the secret?

What about having a `POST earthstar://@xmpl/ {secret: hex string, address}` for importing identities?

Reply:
I like that!  Putting the author and secret in the request headers when posting a new document.  Or let the browser fill in the secret (if the browser is handling secrets).

```
POST earthstar://+workspace.xxxx/hello
body: "world"
headers: {
    "earthstar-author": "@xmpl.xxxxxxxxx"
    "earthstar-secret": "yyyyyyyyy"
}
```

> What about having a `POST earthstar://@xmpl/ {secret: hex string, address}` for importing identities?

Seems ok.  It would really benefit from having a browser UI so ordinary mortals can bring their identities in.